framework/audio/engine/audioenginemodule.cpp

```diff
@@ -0,0 +1,71 @@
+﻿//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "audioenginemodule.h"
+
+#include <QQmlEngine>
+
+#include "modularity/ioc.h"
+#include "internal/audioengine.h"
+
+#include "devtools/audioenginedevtools.h"
+
+#ifdef Q_OS_LINUX
+#include "platform/lin/linuxaudiodriver.h"
+#endif
+
+#ifdef Q_OS_WIN
+#include "platform/win/winaudiodriver.h"
+#endif
+
+using namespace mu::audio::engine;
+
+std::shared_ptr<AudioEngine> audioEngine = std::make_shared<AudioEngine>();
+
+std::string AudioEngineModule::moduleName() const
+{
+    return "audio_engine";
+}
+
+void AudioEngineModule::registerExports()
+{
+    framework::ioc()->registerExport<IAudioEngine>(moduleName(), audioEngine);
+
+#ifdef Q_OS_LINUX
```

Comment:
maybe it would be better to put this in the factory.

Reply:
I think this is unnecessary. What is the point of creating a separate factory, if it will not be used anywhere other than this class, which in any case is the module setup class?